implementations/rust/ockam/ockam_command/src/node/create.rs

```diff
@@ -351,6 +351,20 @@ async fn run_foreground_node(
         stop_node_on_eof(&mut ctx, &opts, &node_name).await?;
     }
```

Comment:
This `stop_node_on_eof` function runs a loop and blocks the execution until it receives an EOF character, so the signal handler should be registered before this block.

We have to support these two loops at the same time and exit on the first one that happens, so we can only block the main thread with only one of the loops.